docs/samples/explanation/aif/germancredit/README.md

```diff
@@ -0,0 +1,85 @@
+
+We would like to be able to check if these "protected classes" are being used in a model's predictions. In this example we will feed the model some predictions and calculate metrics based off of the predictions the model makes. These metrics will give insight as to whether or not the model is biased for or against any protected classes. In this example we will look at the bias our deployed model has on those of age > 25 vs. those of age <= 25 and see if creditors are treating either unfairly.
+
+### Create the InferenceService
+
+Apply the CRD
+
+```
+kubectl apply -f bias.yaml
+```
+
+Expected Output
+
+```
+$ inferenceservice.serving.kubeflow.org/german-credit created
+```
+
+### Deploy the message dumper
+
+Apply the message-dumper CRD which will collect the logs that are created when running predictions on the inferenceservice.
+
+```
+kubectl apply -f message-dumper.yaml
+```
+
+Expected Output
+
+```
+service.serving.knative.dev/message-dumper created
+```
+
+### Run a prediction
+
+The first step is to [determine the ingress IP and ports](../../../../../README.md#determine-the-ingress-ip-and-ports) and set `INGRESS_HOST` and `INGRESS_PORT`
+
+```
+MODEL_NAME=german-credit
+SERVICE_HOSTNAME=$(kubectl get inferenceservice ${MODEL_NAME} -o jsonpath='{.status.url}' | cut -d "/" -f 3)
+python simulate_predicts.py http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/$MODEL_NAME:predict ${SERVICE_HOSTNAME}
+```
+
+### Create a payload from the logs
+
+Run `json_from_logs.py` which will craft a payload that AIF can interpret. First, the events logs are taken from the message-dumper and then those logs are parsed to match inputs with outputs. Then the input/outputs pairs are all combined into a list of inputs and a list of outputs for AIF to interpret. A `data.json` file should have been created in this folder which contains the json payload.
+
+```
+python json_from_logs.py
+```
+
+### Run an explanation
+
+Finally, now that we have collected a number of our model's predictions and their corresponding inputs we will send these to the AIF server to calculate the bias metrics.
+
+```
+python query_bias.py http://${INGRESS_HOST}:${INGRESS_PORT}/v1/models/$MODEL_NAME:explain ${SERVICE_HOSTNAME} data.json
+```
+
+### Interpreting the results
+
+Now lets look at one of the metrics. In this example disparate impact represents the ratio between the probability of applicants of the privileged class (age > 25) getting a loan and the probability of applicants of the unprivileged class (age <= 25) getting a loan `P(Y=1|D=privileged)/P(Y=1|D=unprivileged)`. Since, in the sample output below, the disparate impact is less that 1 then the probability that an applicant whose age is greater than 25 gets a loan is significantly higher than the probability that an applicant whose age is less than or equal to 25 gets a loan. This in and of itself is not proof that the model is biased, but does hint that there may be some bias and a deeper look may be needed.
```

Comment:
```suggestion
Now let's look at one of the metrics. In this example disparate impact represents the ratio between the probability of applicants of the privileged class (age > 25) getting a loan and the probability of applicants of the unprivileged class (age <= 25) getting a loan `P(Y=1|D=privileged)/P(Y=1|D=unprivileged)`. Since, in the sample output below, the disparate impact is less that 1 then the probability that an applicant whose age is greater than 25 gets a loan is significantly higher than the probability that an applicant whose age is less than or equal to 25 gets a loan. This in and of itself is not proof that the model is biased, but does hint that there may be some bias and a deeper look may be needed.
```